drivers/serial/uart_nrfx_uart.c

```diff
@@ -919,9 +930,11 @@ static int uart_nrfx_init(struct device *dev)
 	 */
 	nrf_uart_enable(uart0_addr);
 
-	nrf_uart_event_clear(uart0_addr, NRF_UART_EVENT_RXDRDY);
+	if (IS_RX_PIN()) {
+		nrf_uart_event_clear(uart0_addr, NRF_UART_EVENT_RXDRDY);
 
-	nrf_uart_task_trigger(uart0_addr, NRF_UART_TASK_STARTRX);
+		nrf_uart_task_trigger(uart0_addr, NRF_UART_TASK_STARTRX);
```

Comment:
Same should be done with the STARTRX task triggered in `uart_nrfx_set_power_state`, see [line 1044](https://github.com/zephyrproject-rtos/zephyr/pull/22932/files#diff-24c27a01c8713b6b093dfd80f40883d0R1044).